### PR TITLE
inline JsBuiltIn helper CallInstanceFunction

### PR DESCRIPTION
**Pull request by sigatrev**
JsBuiltIns cannot do ```callback.call(thisArg, ...)``` because ```.call``` is loaded off the function prototype which could be changed a user. CallInstanceFunction performs the same operation, taking the form ```CallInstanceFunction(callback, thisArg, ...)```. This commit results in ```callback``` being directly inlined.

### CODE REVIEW

File: lib/Backend/Inline.cpp [hunk 1]
```diff
@@ -3273,13 +3316,25 @@ Inline::InlineCallTarget(IR::Instr *callInstr, const FunctionJITTimeInfo* inline
...
+    {
+        IR::Instr * functionArg = secondArgOut;
+        IR::Instr * bytecodeArgOutUse = IR::Instr::New(Js::OpCode::BytecodeArgOutUse, callInstr->m_func);
+        callInstr->ReplaceSrc1(functionArg->GetSrc1());
+        bytecodeArgOutUse->SetSrc1(functionArg->GetSrc1());
+        callInstr->InsertBefore(bytecodeArgOutUse); // Need to keep the function argout live till the call instruction for it to be captured by any bailout in the call sequence.
+        functionArg->Remove();
+    }
+
     IR::Instr * bytecodeArgOutUse = IR::Instr::New(Js::OpCode::BytecodeArgOutUse, callInstr->m_func);
```
Comment (MSLaguana) [inline lib/Backend/Inline.cpp]: Is this deliberately duplicated from inside the above conditional? Should it be declared prior to the conditional and conditionally configured, or do we actually want both?

Comment (MSLaguana) [reply lib/Backend/Inline.cpp]: Ah, I see now that they differ on the argument (argument 1 vs argument 2).

File: lib/Common/ConfigFlagsList.h [hunk 2]
```diff
@@ -1074,6 +1075,7 @@ FLAGNR(Boolean, JitRepro              , "Add Function.invokeJit to execute codeg
 FLAGNR(Boolean, EntryPointInfoRpcData , "Keep encoded rpc buffer for jitted function on EntryPointInfo until cleanup", DEFAULT_CONFIG_EntryPointInfoRpcData)
 
 FLAGNR(Boolean, LdChakraLib           , "Access to the Chakra internal library with the __chakraLibrary keyword", DEFAULT_CONFIG_LdChakraLib)
+FLAGNR(Boolean, TestChakraLib         , "Access to the Chakra internal library with the __chakraLibrary keyword without global access restriction", DEFAULT_CONFIG_TestChakraLib)
```
Comment (jackhorton) [inline lib/Common/ConfigFlagsList.h]: 💯

**Comment by sigatrev** [comment]
@jackhorton This will apply to Intl's uses as well, it's the same function just called off of different objects.

**Pull request merged**